### PR TITLE
CSP-860 Include employer and apprentice profile in both outer apis

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MemberProfiles/GetMemberProfileWithPreferencesTests.cs [hunk 1]
```diff
@@ -18,24 +20,26 @@
...
-    [MoqInlineAutoData(false, false)]
+    [MoqInlineAutoData(true, true, MemberUserType.Apprentice)]
+    [MoqInlineAutoData(true, false, MemberUserType.Apprentice)]
+    [MoqInlineAutoData(false, true, MemberUserType.Apprentice)]
+    [MoqInlineAutoData(false, false, MemberUserType.Apprentice)]
+    [MoqInlineAutoData(true, true, MemberUserType.Employer)]
+    [MoqInlineAutoData(true, false, MemberUserType.Employer)]
+    [MoqInlineAutoData(false, true, MemberUserType.Employer)]
+    [MoqInlineAutoData(false, false, MemberUserType.Employer)]
     public async Task When_MediatorCommandSuccessful_Then_ReturnOk(
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MemberProfiles/GetMemberProfileWithPreferencesTests.cs:34]: This test coverage is not sufficient. There are three if conditions in the controller. There should be tests on both sides of these if conditions.

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs [hunk 2]
```diff
@@ -42,7 +42,14 @@ public async Task<GetMemberProfileWithPreferencesQueryResult> Handle(GetMemberPr
         result.UserType = Enum.Parse<MemberUserType>(outputMember.UserType);
 
         result.IsRegionalChair = (bool)outputMember.IsRegionalChair!;
-        result.ApprenticeId = outputMember.Apprentice!.ApprenticeId;
+        if (result.UserType == MemberUserType.Apprentice)
+        {
+            result.ApprenticeId = (outputMember.Apprentice != null) ? outputMember.Apprentice!.ApprenticeId : Guid.Empty;
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs:47]: unnecessary null check

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs [hunk 3]
```diff
@@ -42,7 +42,14 @@ public async Task<GetMemberProfileWithPreferencesQueryResult> Handle(GetMemberPr
...
+        }
+        if (result.UserType == MemberUserType.Employer)
+        {
+            result.AccountId = (outputMember.Employer != null) ? outputMember.Employer!.AccountId : 0;
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs:51]: unnecessary null check

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/SFA.DAS.ApprenticeAan.Application.csproj
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/SFA.DAS.ApprenticeAan.Application.csproj:1]: undo this change

Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/SFA.DAS.ApprenticeAan.Application.csproj:1]: delete empty folder

File: src/EmployerAan/SFA.DAS.EmployerAan/Infrastructure/Constants.cs [hunk 4]
```diff
@@ -6,4 +6,8 @@ public static class ApiHeaders
     {
         public const string RequestedByMemberIdHeader = "X-RequestedByMemberId";
     }
+    public static class Preferences
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/Infrastructure/Constants.cs:9]: Rename to PreferenceIds

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MemberProfiles/GetMemberProfileWithPreferencesTests.cs
[the remark this answers is not in the document]
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MemberProfiles/GetMemberProfileWithPreferencesTests.cs:1]: Done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs (hunk 2 above)
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs:47]: Done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/SFA.DAS.ApprenticeAan.Application.csproj
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/SFA.DAS.ApprenticeAan.Application.csproj:1]: Done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs (hunk 3 above)
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs:51]: Done

File: src/EmployerAan/SFA.DAS.EmployerAan/Infrastructure/Constants.cs (hunk 4 above)
Comment (ManishAgarwalDfE) [reply src/EmployerAan/SFA.DAS.EmployerAan/Infrastructure/Constants.cs]: Done

Comment (ManishAgarwalDfE) [reply src/EmployerAan/SFA.DAS.EmployerAan/Infrastructure/Constants.cs]: Done

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MemberProfiles/GetMemberProfileWithPreferencesTests.cs (hunk 1 above)
Comment (ManishAgarwalDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Controllers/MemberProfiles/GetMemberProfileWithPreferencesTests.cs:34]: I have covered all the three conditions here so not sure what else I need to cover
